safe/reportv4/extractors/analysis_result.py

```diff
@@ -0,0 +1,183 @@
+# coding=utf-8
+import os
+
+from safe.definitionsv4.exposure import exposure_population
+from safe.utilities.i18n import tr
+from safe.definitionsv4.hazard_classifications import generic_hazard_classes
+
+__copyright__ = "Copyright 2016, The InaSAFE Project"
+__license__ = "GPL version 3"
+__email__ = "info@inasafe.org"
+__revision__ = '$Format:%H$'
+
+
+def sum_field_by_hazard(layer, hazard_classification, field_name):
+    """
+
+    :param layer:
+    :type layer: qgis.core.QgsVectorLayer
+    :param hazard_classification:
+    :return: dict
+    """
+    classes = hazard_classification.get('classes')
+    result = []
+    for c in classes:
+        result.append({
+            'key': c.get('key'),
+            'name': c.get('name'),
+            'value': 0
+        })
+
+    hazard_class_idx = layer.fieldNameIndex('hazard_class')
+    target_field_idx = layer.fieldNameIndex(field_name)
+    for f in layer.getFeatures():
+        attrs = f.attributes()
+        try:
+            hazard_class = attrs[hazard_class_idx]
+            idx = [
+                i for i, v in enumerate(result) if v['key'] == hazard_class
+                ][0]
+            result[idx]['value'] += attrs[target_field_idx]
+        except:
+            pass
+
+    return result
+
+
+def count_field_by_hazard(layer, hazard_classification, field_name):
+    """
+
+    :param layer:
+    :type layer: qgis.core.QgsVectorLayer
+    :param hazard_classification:
+    :return: dict
+    """
+    class_name = [c.get('key') for c in hazard_classification.get('classes')]
+    result = {}
+    for c in class_name:
+        result[c] = 0
+
+    hazard_class_idx = layer.fieldNameIndex('hazard_class')
+    for f in layer.getFeatures():
+        attrs = f.attributes()
+        result[attrs[hazard_class_idx]] += 1
+
+    return result
+
+
+def sum_affected_field(layer, hazard_classification, field_name):
+    """
+
+    :param layer:
+    :type layer: qgis.core.QgsVectorLayer
+    :param field_name:
+    :return: dict
+    """
+    hazard_class = hazard_classification.get('classes')
+    affected_dict = {}
+    for c in hazard_class:
+        affected_dict[c.get('key')] = c.get('affected')
+    result = {
+        'affected': 0,
+        'unaffected': 0,
+        'total': 0,
+    }
+    hazard_class_idx = layer.fieldNameIndex('hazard_class')
```

Comment:
Same the name field, we shouldn't hardcode these name field which are defined in definitions.